src/server/rest/v1/service/AssetService.ts

```diff
@@ -238,44 +214,18 @@ export default class AssetService {
     // Check if component is active
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.ASSET,
       Action.RETRIEVE_CONSUMPTION, Entity.ASSET, MODULE_NAME, 'handleRetrieveConsumption');
-    // Is authorized to check connection ?
-    if (!await Authorizations.canRetrieveAssetConsumption(req.user)) {
-      throw new AppAuthError({
-        errorCode: HTTPAuthError.FORBIDDEN,
-        user: req.user,
-        entity: Entity.ASSET, action: Action.RETRIEVE_CONSUMPTION,
-        module: MODULE_NAME, method: 'handleRetrieveConsumption'
-      });
-    }
     // Filter request
     const assetID = AssetValidator.getInstance().validateAssetGetReq(req.query).ID;
     UtilsService.assertIdIsProvided(action, assetID, MODULE_NAME, 'handleRetrieveConsumption', req.user);
+    // Retrieve authorizations for current action attempt
+    const actionOnAssetAuthorizationFilter = await AuthorizationService.checkAndGetAssetAuthorizations(
+      req.tenant, req.user, Action.RETRIEVE_CONSUMPTION, {});
     // Get
-    const asset = await AssetStorage.getAsset(req.tenant, assetID);
-    UtilsService.assertObjectExists(action, asset, `Asset ID '${assetID}' does not exist`,
+    const asset = await AssetService.getAssetFromStorage(req.tenant, assetID, { dynamicOnly: true, usesPushAPI: false }, actionOnAssetAuthorizationFilter);
+    UtilsService.assertObjectExists(action, asset, `Asset ID '${assetID}' consumption cannot be retrieved`,
       MODULE_NAME, 'handleRetrieveConsumption', req.user);
-    // Dynamic asset ?
-    if (!asset.dynamicAsset) {
-      throw new AppError({
-        errorCode: HTTPError.GENERAL_ERROR,
-        module: MODULE_NAME, method: 'handleRetrieveConsumption',
-        action: action,
-        user: req.user,
-        message: 'This Asset is not dynamic, no consumption can be retrieved',
-        detailedMessages: { asset }
-      });
-    }
-    // Uses Push API
-    if (asset.usesPushAPI) {
-      throw new AppError({
-        errorCode: HTTPError.GENERAL_ERROR,
-        module: MODULE_NAME, method: 'handleRetrieveConsumption',
-        action: action,
-        user: req.user,
-        message: 'This Asset is using the push API, no consumption can be retrieved',
-        detailedMessages: { asset }
-      });
-    }
+    // Dynamic auth
+    await AuthorizationService.checkAndGetAssetAuthorizations(req.tenant, req.user, Action.RETRIEVE_CONSUMPTION, {}, asset);
```

Comment:
Check line 264 if you can finish the job :-)
`// TODO: Return a specific HTTP code to tell the user that the consumption cannot be retrieved`
Thanks!